rasa_nlu/evaluate.py

```diff
@@ -502,60 +520,91 @@ def run_cv_evaluation(td, n_folds, nlu_config):
     :return: dictionary with key, list structure, where each entry in list
               corresponds to the relevant result for one fold
     """
-    from sklearn import metrics
     from collections import defaultdict
     import tempfile
 
     trainer = Trainer(nlu_config)
     train_results = defaultdict(list)
     test_results = defaultdict(list)
-
+    entity_train_results = defaultdict(lambda: defaultdict(list))
+    entity_test_results = defaultdict(lambda: defaultdict(list))
     tmp_dir = tempfile.mkdtemp()
 
     for train, test in generate_folds(n_folds, td):
-        trainer.train(TrainingData(training_examples=train,
-                                   entity_synonyms=td.entity_synonyms,
-                                   regex_features=td.regex_features))
+        train = TrainingData(training_examples=train,
```

Comment:
can we just return `TrainingData` instances from `generate_folds` to avoid this duplication?